### PR TITLE
Feature/10/cart page summary, connected to #10

### PR DESCRIPTION
**Pull request by schleichardt**
@lauraluiz please review

preview app:

1. go to http://sphere-sunrise-dmexco-10.herokuapp.com/en/michael-kors-shopper-jetsettravel-30S3GTVT6L-red-A0E200000001YM5.html
1. optional: select a quantity
1. click on "ADD TO BAG"
1. go to http://sphere-sunrise-dmexco-10.herokuapp.com/en/cart
1. if you want to delete the item from the cart, use chrome/resources and delete the Play cookie

### CODE REVIEW

File: app/cart/CartController.java [hunk 1]
```diff
@@ -21,6 +28,7 @@ public CartController(final ControllerDependency controllerDependency) {
         return Optional.ofNullable(session(CartSessionKeys.CART_ID))
                 .map(cartId -> sphere().execute(CartByIdGet.of(cartId)))
                 .orElseGet(() -> sphere().execute(CartCreateCommand.of(CartDraft.of(userContext.currency()).withCountry(userContext.country())))
+                        .flatMap(cart -> sphere().execute(CartUpdateCommand.of(cart, SetShippingAddress.of(DEFAULT_ADDRESS))))
```
Comment (schleichardt) [inline app/cart/CartController.java]: Especially here I am not confident if I should already set the shipping address. But without the cart detail page would display no taxes.

File: app/cart/CartItems.java [hunk 2]
```diff
@@ -1,15 +1,55 @@
...
 
+    public static CartItems of(final Cart cart, final UserContext userContext) {
+        final CartItems cartItems = new CartItems();
+        final List<CartItem> cartItemList = cart.getLineItems()
+                .stream()
+                .map(lineItem -> CartItem.of(lineItem, userContext))
+                .collect(toList());
+        cartItems.setList(cartItemList);
+        final PriceFormatter priceFormatter = PriceFormatter.of(userContext.locale());
+        final String subtotal = priceFormatter.format(cart.getTotalPrice());
```
Comment (lauraluiz) [inline app/cart/CartItems.java]: subtotal = sum(all line items)
Without cart discounts and shipping costs

File: app/cart/CartItems.java [hunk 3]
```diff
@@ -1,15 +1,55 @@
...
+                .map(cartShippingInfo -> priceFormatter.format(cartShippingInfo.getPrice()))
+                .orElse(zeroMoneyAmount);
+        cartItems.setDelivery(delivery);
+        final String tax = Optional.ofNullable(cart.getTaxedPrice())
+                .map(taxedPrice -> taxedPrice.getTotalGross().subtract(taxedPrice.getTotalNet()))
+                .map(amount -> priceFormatter.format(amount))
+                .orElse(zeroMoneyAmount);
+        cartItems.setTax(tax);
+        final String orderTotal = Optional.ofNullable(cart.getTaxedPrice())
+                .map(taxedPrice -> priceFormatter.format(taxedPrice.getTotalNet()))
```
Comment (lauraluiz) [inline app/cart/CartItems.java]: `getTotalGross`

File: app/cart/CartController.java [hunk 4]
```diff
@@ -1,18 +1,25 @@
...
 import io.sphere.sdk.carts.queries.CartByIdGet;
+import io.sphere.sdk.models.Address;
 import play.libs.F;
 import play.mvc.Http;
 
 import java.util.Optional;
 
 public abstract class CartController extends SunriseController {
+
+    private static final Address DEFAULT_ADDRESS = Address.of(CountryCode.DE);
```
Comment (lauraluiz) [inline app/cart/CartController.java]: `UserContext.country`

File: app/cart/CartController.java (hunk 1 above)
Comment (lauraluiz) [inline app/cart/CartController.java]: In Europe it is always normal to predefine the shipping to calculate taxes, so it's good :+1:

File: app/cart/CartItems.java [hunk 5]
```diff
@@ -1,15 +1,55 @@
...
+                .orElse(subtotal);
```
Comment (lauraluiz) [inline app/cart/CartItems.java]: Here you can use `getTotalPrice` once you change how subtotal is calculated.

**Comment by schleichardt** [comment]
@lauraluiz I updated it: http://sphere-sunrise-dmexco-10.herokuapp.com/en/cart

File: app/cart/CartItems.java [hunk 6]
```diff
@@ -1,15 +1,72 @@
...
+                .orElse(zeroMoneyAmount);
+        cartItems.setDelivery(delivery);
+        final String tax = Optional.ofNullable(cart.getTaxedPrice())
+                .map(taxedPrice -> taxedPrice.getTotalGross().subtract(taxedPrice.getTotalNet()))
+                .map(amount -> priceFormatter.format(amount))
+                .orElse(zeroMoneyAmount);
+        cartItems.setTax(tax);
+        final String orderTotal = Optional.ofNullable(cart.getTaxedPrice())
+                .map(taxedPrice -> priceFormatter.format(taxedPrice.getTotalGross()))
+                .orElse(subtotal);
```
Comment (lauraluiz) [inline app/cart/CartItems.java]: Here you can use getTotalPrice, it will give you more accurate information than subtotal.

**Comment by lauraluiz** [comment]
Check the `getTotalPrice` instead of `subtotal`. Besides that, :+1:

**Pull request merged**